pkg/kubelet/preemption/preemption.go

```diff
@@ -199,12 +200,26 @@ func (a admissionRequirementList) distance(pod *v1.Pod) float64 {
 
 // subtract returns a new admissionRequirementList containing remaining requirements if the provided pod
 // were to be preempted
-func (a admissionRequirementList) subtract(pods ...*v1.Pod) admissionRequirementList {
+func (a admissionRequirementList) subtract(cache map[string]int64, pods ...*v1.Pod) admissionRequirementList {
 	newList := []*admissionRequirement{}
 	for _, req := range a {
 		newQuantity := req.quantity
 		for _, pod := range pods {
-			newQuantity -= resource.GetResourceRequest(pod, req.resourceName)
+			if cache != nil {
+				key := pod.GetSelfLink()
```

Comment:
Nit: If we do move forward with this idea, it would be nice to extract the code that either retrieves the value from the cache or fetches it and populates the cache into a helper function.

Reply:
Sure.
After I receive more reviews, I can extract this into a func.